src/Stratis.Bitcoin.Features.WatchOnlyWallet/WatchOnlyWalletManager.cs

```diff
@@ -117,6 +127,68 @@ public void ProcessTransaction(Transaction transaction, Block block = null)
             var transactionHash = transaction.GetHash();
             this.logger.LogDebug($"watch only wallet received transaction - hash: {transactionHash}, coin: {this.coinType}");
 
+            // Check the transaction inputs to see if a watched address is affected.
+            foreach (TxIn input in transaction.Inputs)
+            {
+                // See if the previous transaction is in the watch-only wallet.
+                this.txLookup.TryGetValue(input.PrevOut.Hash, out TransactionData prevTransaction);
+
+                // If it is null, it can't be related to one of the watched addresses (or it is the very first watched transaction)
+                if (prevTransaction == null)
+                    continue;
+
+                // Check if the previous transaction's outputs contain one of our addresses.
+                foreach (TxOut prevOutput in prevTransaction.Transaction.Outputs)
+                {
+                    this.Wallet.WatchedAddresses.TryGetValue(prevOutput.ScriptPubKey.ToString(), out WatchedAddress addressInWallet);
+
+                    if (addressInWallet != null)
+                    {
+                        // Retrieve a transaction, if present.
+                        addressInWallet.Transactions.TryGetValue(transactionHash.ToString(), out TransactionData existingTransaction);
+
+                        if (existingTransaction == null)
+                        {
+                            TransactionData newTransaction = new TransactionData
+                            {
+                                Id = transactionHash,
+                                Hex = transaction.ToHex(),
+                                BlockHash = block?.GetHash(),
+                            };
+
+                            // Add the Merkle proof to the transaction.
+                            if (block != null)
+                            {
+                                newTransaction.MerkleProof = new MerkleBlock(block, new[] { transactionHash }).PartialMerkleTree;
+                            }
+
+                            addressInWallet.Transactions.TryAdd(transactionHash.ToString(), newTransaction);
+
+                            // Update the lookup cache with the new transaction information.
+                            this.txLookup.Remove(newTransaction.Id);
```

Comment:
How can it be that a trx is updated? (what can be updated)

Reply:
If a transaction is added to the watch only wallet, it may be in an unconfirmed state initially (i.e. there is no block hash and no Merkle proof). When the transaction is later confirmed, these details get amended.